sql/codegen.go

```diff
@@ -238,39 +252,70 @@ for name in feature_column_names:
 Y = columns[field_names.index("{{.Y.Name}}")]
 {{- end}}
 
-classifier = tf.estimator.{{.Estimator}}(
-    feature_columns=feature_columns,{{range $key, $value := .Attrs}}
-    {{$key}} = {{$value}},{{end}}
-    model_dir = "{{.Save}}")
+classifier = {{.Estimator}}(
+	feature_columns=feature_columns,{{if .SelfDefined}}{{range $key, $value := .Attrs}}
+	{{$key}} = {{$value}},{{end}})
+	{{else}}{{range $key, $value := .Attrs}}
+	{{$key}} = {{$value}},{{end}}
+	model_dir = "{{.Save}}")
```

Comment:
Suggested feature, maybe adding in the future PR:
Is it possible to add `model_dir` parameter to our self-defined model? So that `my_model.h5` will be stored at `model_dir` as `modir_dir/my_model.h5`. Later on, if we need to save training logs and visualize ten, we can also save them at `model_dir`.

Reply:
Created an issue: https://github.com/sql-machine-learning/sqlflow/issues/432 to track this.